src/core/server/http/http_service.ts

```diff
@@ -20,15 +20,18 @@
 import { Observable, Subscription } from 'rxjs';
 import { first, map } from 'rxjs/operators';
 
+import { LoggerFactory } from '../logging';
 import { CoreService } from '../../types';
 import { Logger } from '../logging';
 import { CoreContext } from '../core_context';
-import { HttpConfig, HttpConfigType } from './http_config';
+import { HttpConfig, HttpConfigType, config as httpConfig } from './http_config';
 import { HttpServer, HttpServerSetup } from './http_server';
 import { HttpsRedirectServer } from './https_redirect_server';
 
 /** @public */
-export type HttpServiceSetup = HttpServerSetup;
+export type HttpServiceSetup = HttpServerSetup & {
+  createNewServer: (cfg: Partial<HttpConfig>) => Promise<HttpServerSetup>;
```

Comment:
**nit** it is public interface and `cfg` doesn't tell much for end users, I'd rather declare here a full name. if we have a problem with name-shadowing [here](https://github.com/elastic/kibana/pull/36804/files#diff-f6edd5d99b796ea3adfdd32e2291ca1eR81) we can re-write it as 
```js
...{ createNewServer: this.createServer.bind(this) },
```